docs/design/crypto-transfer.md

```diff
@@ -0,0 +1,78 @@
+
+# Atomic CryptoTransfer
+
+## Purpose
+
+The existing Crypto transfer precompile functionality is missing the ability to atomically transfer `hbars` along with fungible and non-fungible tokens.  In addition, there is no support for using the approval mechanism.
+
+The Crypto transfer precompile should be refactored to include this functionality in order to better align with HAPI.
+
+This document details how to refactor the existing code in order to gain the additional functionality desired.
+
+## Goals
+
+When performing a Crypto transfer precompile:
+- Allow for atomic transfers of hbars, fungible tokens and non-fungible tokens between accounts.
+- Support `approval` functionality with transfers.
+## Non Goals
+Continued support for the existing interface for the `cryptoTransfer` precompile.  While support for the existing functionality will continue in the Hedera services layer for backwards compatibility, `IHederaTokenService` will be updated to only support the new function call.
+## Architecture
+Update the information passed into the `cryptoTransfer` precompile in order to allow for `hbar` transfers and `approval` functionality.  Refactor the Crypto transfer precompile in Hedera services to perform the transfers specified in the input parameters
+## Non-Functional Requirements
+
+### Hedera Smart Contract Changes
+
+#### Update  `AccountAmount`  struct to support  `isApproval`
+
+    struct AccountAmount {
```

Comment:
When we adjust the solidity file we should fine some place in docs to notate version differences, and bump the major version as this introduces incompatibilities.

Reply:
Probably `@custom:version` - https://docs.soliditylang.org/en/develop/natspec-format.html#tags

Reply:
And we may be able to do just a minor version bump instead of a major version per semver if we are 0.x.y  still - https://semver.org/#spec-item-4